### PR TITLE
Lint for more styleguide errors

### PR DESCRIPTION
**Pull request by mnquintana**
This PR adds more coffeelint rules to cover code style requirements from CONTRIBUTING.md, and one rule that wasn't explicitly defined in CONTRIBUTING.md, but was already used everywhere in the codebase.

Refs atom/settings-view#449, atom/settings-view#44, #5982 

/cc @kevinsawicki I wanted to update the coffeelint rules first in core before I go and add coffeelint.json to all the package repos.

### CODE REVIEW

**Comment by benogle** [comment]
Note that the colon spacing rule will annoyingly fail in a regex containing
a colon.

**Comment by mnquintana** [comment]
@benogle Aww :cry: – alright then, I'll take that one out. How does it look otherwise?

**Comment by mnquintana** [comment]
@benogle Is that still the case? I tested it with this regex:

```coffeescript
/[A-Za-z0-9.,:-]*/
```

And didn't get any coffeelint errors.

**Comment by benogle** [comment]
May have been a different version of coffee lint. I had problems in the
autocomplete linter. If it doesn't throw an error, carry on.

**Comment by kevinsawicki** [comment]
I like these changes but the build will fail if this is merged since there are cases currently where these aren't being followed.

Did you want to fix these in this pull request or just change these additions to warnings to start?

**Comment by mnquintana** [comment]
@kevinsawicki I was planning on fixing the code style errors in this PR; I just wanted to make sure the coffeelint changes looked good first. If it all looks good then I'll take care of that when I get home!

**Comment by kevinsawicki** [comment]
>  If it all looks good then

Yup, looks good to me :+1:

**Comment by mnquintana** [comment]
@kevinsawicki This should be ready for review now. (How do you trigger CI builds? I can't tell if it's passing or not.)

File: spec/fixtures/coffee.coffee [hunk 1]
```diff
@@ -13,6 +13,7 @@ class quicksort
       if current < pivot
         left.push(current)
       else
+        # coffeelint: disable=no_trailing_semicolons
```
Comment (kevinsawicki) [inline spec/fixtures/coffee.coffee]: I think we probably want to omit all `spec/fixtures` files from any linting.

I'm guessing adding this line might cause some specs to fail that may be targeting specific lines in this file by number.

**Comment by kevinsawicki** [comment]
> How do you trigger CI builds?

CI builds only run for local branches currently, not branches from forks, sorry, this is being fixed in #6222  though

File: spec/fixtures/coffee.coffee (hunk 1 above)
Comment (mnquintana) [inline spec/fixtures/coffee.coffee]: That makes sense - we could handle that by adding a `.coffeelintignore` (oh how I wish we could just configure it in `coffeelint.json`), but that would be redundant with the coffeelint grunt task config. So I could just revert the changes to fixtures, if that would work?

Comment (kevinsawicki) [inline spec/fixtures/coffee.coffee]: > So I could just revert the changes to fixtures, if that would work?

That sounds good to me

**Comment by mnquintana** [comment]
@kevinsawicki This should be ready now! (I kept `spec/fixtures/packages/package-with-config-defaults/index.coffee` as is though because it shouldn't do any harm)

File: benchmark/fixtures/medium.coffee [hunk 2]
```diff
@@ -209,13 +209,13 @@ template = (str) ->
...
-       .split('<%').join("');")
-       .split('%>').join("p.push('") +
-       "');}return p.join('');"
+      .replace(/'(?=[^<]*%>)/g,"\t")
+      .split("'").join("\\'")
+      .split("\t").join("'")
+      .replace(/<%=(.+?)%>/g, "',$1,'")
+      .split('<%').join("');")
+      .split('%>').join("p.push('") +
+      "');}return p.join('');"
```
Comment (kevinsawicki) [inline benchmark/fixtures/medium.coffee]: Mind backing this out since it is a fixture as well?

File: spec/fixtures/packages/package-with-config-defaults/index.coffee [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module.exports =
   configDefaults:
-    numbers: { one: 1, two: 2 }
+    numbers: {one: 1, two: 2}
```
Comment (kevinsawicki) [inline spec/fixtures/packages/package-with-config-defaults/index.coffee]: Also mind reverting the changes to this fixture as well?

**Comment by mnquintana** [comment]
@kevinsawicki Addressed the comments!

**Comment by kevinsawicki** [comment]
Awesome, thanks for cleaning these all up :green_heart:

**Comment by thomasjo** [comment]
Thanks for doing this @mnquintana! :sparkling_heart:
